compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -0,0 +1,1394 @@
+                this.env.pkgSymbol.pkgID, funcType, this.env.pkgSymbol, Symbols.isFlagOn(flags, Flags.NATIVE));
+        Scope scopeToDefine = this.env.pkgSymbol.scope;
+
+//        if (Symbols.isFlagOn(flags, Flags.ATTACHED)) {
+//            int attachedToTypeRefCPIndex = dataInStream.readInt();
+//            TypeRefCPEntry typeRefCPEntry = (TypeRefCPEntry) this.env.constantPool[attachedToTypeRefCPIndex];
+//            UTF8CPEntry typeSigCPEntry = (UTF8CPEntry) this.env.constantPool[typeRefCPEntry.typeSigCPIndex];
+//            BType attachedType = getBTypeFromDescriptor(typeSigCPEntry.getValue());
+//
+//            // Update the symbol
+//            invokableSymbol.owner = attachedType.tsymbol;
+//            invokableSymbol.name =
+//                    names.fromString(Symbols.getAttachedFuncSymbolName(attachedType.tsymbol.name.value, funcName));
+//            if (attachedType.tag == TypeTags.OBJECT || attachedType.tag == TypeTags.RECORD) {
+//                if (attachedType.tag == TypeTags.OBJECT) {
+//                    scopeToDefine = ((BObjectTypeSymbol) attachedType.tsymbol).methodScope;
+//                } else {
+//                    scopeToDefine = attachedType.tsymbol.scope;
+//                }
+//                BAttachedFunction attachedFunc =
+//                        new BAttachedFunction(names.fromString(funcName), invokableSymbol, funcType);
+//                BStructureTypeSymbol structureTypeSymbol = (BStructureTypeSymbol) attachedType.tsymbol;
+//                structureTypeSymbol.attachedFuncs.add(attachedFunc);
+//                if (Names.OBJECT_INIT_SUFFIX.value.equals(funcName)
+//                        || funcName.equals(Names.INIT_FUNCTION_SUFFIX.value)) {
+//                    structureTypeSymbol.initializerFunc = attachedFunc;
+//                }
+//            }
+//        }
+
+        // set parameter symbols to the function symbol
+        setParamSymbols(invokableSymbol, dataInStream);
+//
+//        // set taint table to the function symbol
+//        setTaintTable(invokableSymbol, attrDataMap);
+//
+//        setDocumentation(invokableSymbol, attrDataMap);
+
+
+        dataInStream.skip(dataInStream.readLong()); // read and skip method body
+
+        if (this.defineFuncsInPkg) { //TODO find a better way?
+            scopeToDefine.define(invokableSymbol.name, invokableSymbol);
+        }
+    }
+
+    private void skipPosition(DataInputStream dataInStream) throws IOException {
+        // TODO find a better way to skip this
+        dataInStream.readInt();
+        dataInStream.readInt();
+        dataInStream.readInt();
+        dataInStream.readInt();
+        dataInStream.readInt();
+    }
+
+    private int visibilityAsMask(int flags, byte visibility) {
+        switch (visibility) {
+            case 0:
+                return flags;
+            case 1:
+                return flags | Flags.PRIVATE;
+            case 2:
```

Comment:
shall we use constants?

Reply:
+1 shall we do that in a separate PR? basically if no conceptual issue, shall we merge this? because I'm anyway working on code cleaning and improving the symbol enter, will improve these with that